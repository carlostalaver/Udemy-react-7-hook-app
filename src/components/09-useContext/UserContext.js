
/* Notar que el nombre del archivo empieza con MAYUCULAS porq basicamente un context es basicamente un context
   componente
*/
import { createContext }  from 'react';


/* este UserContext es un componente de orden superior, es decir, es un componente que tiene HIJOS */
export const UserContext =  createContext(null);
