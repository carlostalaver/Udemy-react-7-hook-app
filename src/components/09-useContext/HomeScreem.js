import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const HomeScreem = () => {

    /* De todos los contexts que pudiera tener mi app, solicito el UserContext */
   const { user } = useContext(UserContext);

   console.log(user);

    return (
        <div>
            <h1>HomeScreem</h1>
            <hr/>

            <pre className=" container ">
               { JSON.stringify(user, null, 3) }
            </pre>
            
        </div>
    )
}
