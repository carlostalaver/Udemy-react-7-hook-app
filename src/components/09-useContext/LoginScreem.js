import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreem = () => {
    const { user, setUser } = useContext(UserContext);


    return (
        <div>
            <h1>LoginScreem</h1>
            <hr/>

            <button  className="btn btn-primary" onClick= { () => setUser({ id: 1234, name: 'carlos'}) }>
                Login
            </button>
            
        </div>
    )
}