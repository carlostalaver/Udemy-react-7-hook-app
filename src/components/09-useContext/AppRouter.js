import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { AboutScreem } from './AboutScreem';
import { HomeScreem } from './HomeScreem';
import { LoginScreem } from './LoginScreem';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className="container">

                    <Switch>
                        {/* exact={ true } para que la URL sea exacta, tambien se puede colocar exact sin = { true } */}
                        <Route exact={ true } path="/about" component={AboutScreem} />
                        <Route exact={ true } path="/login" component={LoginScreem} />
                        <Route exact={ true } path="/" component={HomeScreem} />
                        <Redirect to="/" />  {/* Ruta por defecto, notar que no tiene la palabre exact */}
                    </Switch>

                </div>
            </div>
        </Router>
    )
}
