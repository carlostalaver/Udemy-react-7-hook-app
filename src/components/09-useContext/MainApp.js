import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'


export const MainApp = () => {

        const [user, setUser]  = useState({});
    /* value= { user } hará que el obj user esté disponible para todos los hijos, nietos, bisnietos... 
       de  UserContext.Provider, para obtener el valor en uno de los hijos se usa el hook useContext 
       pasandole como parametro mi contexto, es decir, el UserContext, ver HomeScreem.js
       */
    return (
        /* Ojo que UserContext es mi componente de orden superior que definí en UserContext.js.
            {user, setUser}: es un objeto literal que le paso en la propiedad value.
            cualquier cambio que se haga en el objeto user lo pillará UserContext y comunicará dichos cambios
            al resto de los hijos que hagan uso de este contexto
        */
        <UserContext.Provider value= { {user, setUser} }>
            <AppRouter />
        </UserContext.Provider>
    )
}
