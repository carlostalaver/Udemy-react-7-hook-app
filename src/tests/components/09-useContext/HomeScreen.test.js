import { React } from 'react';
import { renderHook, act } from '@testing-library/react-hooks'
import { shallow, mount } from 'enzyme';
import { TodoApp } from '../../../components/08-useReducer/TodoApp';
import { demoTodos } from '../../fixtures/demoTodos';
import { HomeScreem } from '../../../components/09-useContext/HomeScreem';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en <HomeScreen/>', () => {

    const user = {
        name:'Carlos',
        email: 'carlos@example.com',
    };

    /* Tengo que usar mount en lugar de shallow porque necesito el componente nieto (<HomeScreem />).
        <UserContext.Provider value= {{
            user: user
        }}></UserContext.Provider>

        se lo proporciono tal cual lo definí porq de lo contrario fallara la prueba xq solo redireccionará  <UserContext.Provider></UserContext.Provider>
    */
    const wrapper = mount( 
        <UserContext.Provider value= {{
            user: user
        }}>
            <HomeScreem />
        </UserContext.Provider>
     );


    test('debe de mostrarse correctamente ', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    



})