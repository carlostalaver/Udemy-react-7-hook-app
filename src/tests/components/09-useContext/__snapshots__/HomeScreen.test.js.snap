// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    HomeScreem
  </h1>
  <hr />
  <pre
    className=" container "
  >
    {
   "name": "Carlos",
   "email": "carlos@example.com"
}
  </pre>
</div>
`;
